src/editor.rs

```diff
@@ -278,6 +280,24 @@ impl Editor {
             .into(),
         );
 
+        //grey out empty, malformed, and inactive instructions
+        {
+            let mut editor = self.0.borrow_mut();
+            for line in editor.component.iter_mut() {
```

Comment:
The InstrInfo is a direct function of the line contents, so I think we can do this without iterating over every line in `on_change` -- it seems like this should be bound directly to whenever LineInfo is changed (which currently happens in push_line and replace_range).